### PR TITLE
URIBuilder and URIUtils improvements 

### PR DESCRIPTION
**Pull request by ok2c**

### CODE REVIEW

File: httpclient-cache/src/main/java/org/apache/http/impl/client/cache/CacheKeyGenerator.java [hunk 1]
```diff
@@ -63,45 +114,23 @@
      * @return String the extracted URI
      */
     public String getURI(final HttpHost host, final HttpRequest req) {
-        if (isRelativeRequest(req)) {
-            return canonicalizeUri(String.format("%s%s", host.toString(), req.getRequestLine().getUri()));
+        try {
+            final URI uri = normalize(getRequestUri(req, host));
```
Comment (cstamas) [inline httpclient-cache/src/main/java/org/apache/http/impl/client/cache/CacheKeyGenerator.java]: Why `uri` local var?

Comment (ok2c) [reply httpclient-cache/src/main/java/org/apache/http/impl/client/cache/CacheKeyGenerator.java]: @cstamas No reason. Just old habit. It is not a deal breaker, is it?

Comment (cstamas) [reply httpclient-cache/src/main/java/org/apache/http/impl/client/cache/CacheKeyGenerator.java]: Nope, is fine, I just found it strange 😄

Comment (ok2c) [reply httpclient-cache/src/main/java/org/apache/http/impl/client/cache/CacheKeyGenerator.java]: @cstamas I cant help it. `return normalize(getRequestUri(req, host)).toASCIIString();` just looks ugly to me.

Comment (cstamas) [reply httpclient-cache/src/main/java/org/apache/http/impl/client/cache/CacheKeyGenerator.java]: Agreed

**Pull request merged**